haskell-library/flake.nix

```diff
@@ -0,0 +1,111 @@
+{
+  nixConfig = {
+    flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
+  };
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    flake-compat = {
+      flake = false;
+    };
+    haskell-nix = {
+      inputs.hackage.follows = "hackage";
+      inputs.stackage.follows = "stackage";
+    };
+    hackage = {
+      flake = false;
+    };
+    stackage = {
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, serokell-nix, flake-compat, flake-utils, ... }:
+  flake-utils.lib.eachDefaultSystem(system:
+    let
+      haskellPkgs = haskell-nix.legacyPackages."${system}";
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [
+          serokell-nix.overlay
+        ];
+      };
+
+      lib = pkgs.lib;
+
+      hs-package-name = "pataq-package";
+
+      ghc-versions = [ "884" "8107" "902" ];
+
+      # invoke haskell.nix for each ghc version listed in ghc-versions
+       pkgs-per-ghc = lib.genAttrs (map (v: "ghc${v}") ghc-versions)
+        (ghc: haskellPkgs.haskell-nix.cabalProject {
+          src = haskellPkgs.haskell-nix.haskellLib.cleanGit {
+            name = hs-package-name;
+            src = ./.;
+          };
+          compiler-nix-name = ghc;
+
+        # haskell.nix configuration
+          modules = [{
+            packages.${hs-package-name} = {
+            ghcOptions = [
+              # fail on warnings
+              "-Werror"
+              # disable optimisations, we don't need them if we don't package or deploy the executable
+              "-O0"
+            ];
+          };
+
+        }];
+        });
+
+      # returns the list of all components for a package
+      get-package-components = pkg:
+        # library
+        lib.optional (pkg ? library) pkg.library
+        # haddock
+        ++ lib.optional (pkg ? library) pkg.library.haddock
+        # exes, tests and benchmarks
+        ++ lib.attrValues pkg.exes
+        ++ lib.attrValues pkg.tests
+        ++ lib.attrValues pkg.benchmarks;
+
+      # add a prefix to the component name
+      add-name-prefix = prefix: pkg: pkg.overrideAttrs(old:{
+        name = "${prefix}-${old.name or old.pname}";
+      });
+
+      # all components for each specified ghc version
+      all-components = lib.mapAttrs (ghc: pkg:
+        let components =  map (add-name-prefix ghc)
+          (get-package-components pkg.${hs-package-name}.components);
+        in pkgs.linkFarmFromDrvs "all-components" components) pkgs-per-ghc;
+
+      # all tests for each specified ghc version
+      all-tests = lib.mapAttrs (ghc: pkg:
+        let test = pkg.${hs-package-name}.checks."${hs-package-name}-test";
```

Comment:
I think it's safer to iterate over `pkg.${hs-package-name}.checks` explicitly and avoid relying on the fact that `"${hs-package-name}-test" test-suite exists

Reply:
The same should probably be done for the application template :thinking: